src/commands/deal/deploy.ts

```diff
@@ -162,8 +142,19 @@ export default class Deploy extends BaseCommand<typeof Deploy> {
       definition: appCID,
     } of uploadResult.workers) {
       const deal = fluenceConfig.deals?.[workerName];
-      assert(deal !== undefined);
-      const { minWorkers = MIN_WORKERS, targetWorkers = TARGET_WORKERS } = deal;
+
+      assert(
+        deal !== undefined,
+        "Unreachable. All worker names are checked above",
+      );
```

Comment:
I don't like this assert. If you made validation already, probably your validating function should return refined `fluenceConfig.deals` like a proof of validation? Probably every validation should either throw or return refined value e. g. without union with `undefined` or `null`.